contracts/AuctionBase.sol

```diff
@@ -215,7 +202,23 @@ contract AuctionBase is Pausable {
   /// @dev Returns true if the NFT is on auction.
   /// @param _auction - Auction to check.
   function _isActive(Auction storage _auction) internal view returns (bool) {
-    return (_auction.startedAt > 0);
+    return (_auction.startedAt > 0 && !_auction.cancelled && _getAuctionEndAt(_auction) <= now);
+  }
+
+  function _isEnded(Auction storage _auction) internal view returns (bool) {
```

Comment:
Is there a performance hit from passing around a struct as arguments? I haven't seen other solidity code do this.

Reply:
Not sure, I assumed you guys knew what you were doing for stuff like `_isActive`. Can check

Reply:
So, it's apparently okay to use it. Using `storage` is not good. Also, using structs may sometimes cause bugs depending on the fields in the struct (a la nested structs). Regardless, I will create an issue and change this along with #17